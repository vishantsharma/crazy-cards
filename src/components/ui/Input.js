import React from 'react';

const Input = ({name, labelName, type, value, onChange, placeholder, ...props}) => {
    return (
        <>
            <label htmlFor={name}>{labelName}</label>
            <input type={type}
                id={name}
                name={name}
                value={value}
                onChange={onChange}
                placeholder={placeholder}
                {...props}
                /> 
        </>
        )
}

export default Input;