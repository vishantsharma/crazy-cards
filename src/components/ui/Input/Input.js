import React from 'react';

const Input = ({ name, labelName, type, value, onChange, placeholder, error, ...props }) => {
    return (
        <>
            <label htmlFor={name}>{labelName}</label>
            {error ? (
                <div role="alert" className="caution">
                    <span>{error}</span>
                </div>
            ) : null}
            <input type={type}
                id={name}
                name={name}
                value={value}
                onChange={onChange}
                placeholder={placeholder}
                {...props}
            />
        </>
    )
}

export default Input;