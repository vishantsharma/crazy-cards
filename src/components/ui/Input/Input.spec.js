import React from 'react';
import TestRenderer from 'react-test-renderer';
import Input from './Input';

describe('Input', () => {
    let props = {
        name: '', 
        labelName: '', 
        type: '', 
        value: '', 
        placeholder: '', 
        error: '',
        onChange: jest.fn().mockReturnValue()
    }
  it('should render', () => {
    const component = TestRenderer.create(<Input {...props} />)
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  })
})