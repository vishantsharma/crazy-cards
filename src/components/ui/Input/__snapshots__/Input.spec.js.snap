// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should render 1`] = `
Array [
  <label
    htmlFor=""
  >
    
  </label>,
  <input
    id=""
    name=""
    onChange={[MockFunction]}
    placeholder=""
    type=""
    value=""
  />,
]
`;
